cmd/launcher/watchdog/controller_windows.go

```diff
@@ -0,0 +1,340 @@
+		// the main launcher service has attempted to start anyway
+		DelayedAutoStart: true,
+	}
+
+	serviceArgs := []string{"watchdog"}
+	// add any original service arguments from the main launcher service invocation (currently running)
+	// this is likely just a pointer to the launcher.flags file but we want to ensure that the watchdog service
+	// has insight into the same options for early service configuration, logging, etc.
+	serviceArgs = append(serviceArgs, os.Args[2:]...)
+
+	restartService, err := serviceManager.CreateService(
+		launcherWatchdogServiceName,
+		currentExe,
+		svcMgrConf,
+		serviceArgs...,
+	)
+
+	if err != nil { // no point moving forward if we can't create the service
+		return err
+	}
+
+	defer restartService.Close()
+
+	// set recovery actions - always restart after a 5 second delay
+	recoveryActions := []mgr.RecoveryAction{
+		{
+			Type:  mgr.ServiceRestart,
+			Delay: 5 * time.Second,
+		},
+	}
+
+	if err = restartService.SetRecoveryActions(recoveryActions, 10800); err != nil {
+		wc.slogger.Log(ctx, slog.LevelWarn,
+			"unable to set recovery actions for service installation, proceeding",
+			"err", err,
+		)
+	}
+
+	// set recovery actions on non crash failures - indicates that we want service manager
+	// to restart this service after terminating without a state of SERVICE_STOPPED, or whenever
+	// the exit code is not 0 (ERROR_SUCCESS)
+	if err = restartService.SetRecoveryActionsOnNonCrashFailures(true); err != nil {
+		wc.slogger.Log(ctx, slog.LevelWarn,
+			"unable to set RecoveryActionsOnNonCrashFailures flag, proceeding",
+			"err", err,
+		)
+	}
+
+	if err = restartService.Start(); err != nil {
+		wc.slogger.Log(ctx, slog.LevelWarn,
+			"unable to start launcher restart service",
+			"err", err,
+		)
+	}
+
+	wc.slogger.Log(ctx, slog.LevelInfo, "completed watchdog service installation")
+
+	return nil
+}
+
+// removeService utilizes the passed serviceManager to remove the existing service
+// after looking up the handle from serviceName
+func removeService(serviceManager *mgr.Mgr, serviceName string) error {
+	existingService, err := serviceManager.OpenService(serviceName)
+	if err != nil {
+		return err
+	}
+
+	defer existingService.Close()
+
+	if err = existingService.Delete(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (wc *WatchdogController) restartService(service *mgr.Service) error {
+	status, err := service.Control(svc.Stop)
+	if err != nil {
+		wc.slogger.Log(context.TODO(), slog.LevelWarn,
+			"error stopping service",
+			"err", err,
+		)
+
+		// always attempt to start the service regardless, if the service was already
+		// stopped it will still err on the control (stop) call above
+		return service.Start()
+	}
+
+	if err := backoff.WaitFor(func() error {
+		status, err = service.Query()
+		if err != nil {
+			return fmt.Errorf("could not retrieve service status: %w", err)
+		}
+
+		if status.State != svc.Stopped {
+			return fmt.Errorf("service has not stopped")
```

Comment:
```suggestion
			return fmt.Errorf("service has not stopped: state %d", status.State)
```

or `errors.New("service has not stopped")`, but I feel like it might be helpful to know what state it's stuck in?